### PR TITLE
[spotify-web-playback-sdk] add types for PlaybackState context metadata

### PR DESCRIPTION
**Pull request by antoniaelsen**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [ X [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes:  
https://developer.spotify.com/documentation/web-playback-sdk/reference/#object-web-playback-state

Note: the metadata object isn't explicitly typed in the documentation. I've typed it after inspecting the response from the player. If that's not sufficient, no worries -- can close this.

### CODE REVIEW

**Review by mraerino — COMMENTED** [review]
can you add a full example object of the playback context in the test?
e.g. just copy the example data from the docs: https://developer.spotify.com/documentation/web-playback-sdk/reference/#object-web-playback-state

**Comment by aelsen** [comment]
Thanks mraerino --added a couple for context and state, pulled from a real event, because the API documentation doesn't go into as much detail. And -- found that some things were out of date there too.

File: types/spotify-web-playback-sdk/index.d.ts [hunk 1]
```diff
@@ -99,6 +121,11 @@ declare namespace Spotify {
         shuffle: boolean;
         restrictions: PlaybackRestrictions;
         track_window: PlaybackTrackWindow;
+        playback_id: string;
+        playback_quality: string;
+        playback_features: {
+            hifi_status: string;
```
Comment (mraerino) [inline types/spotify-web-playback-sdk/index.d.ts:127]: do we know the possible values for this? feels like an enum to me

File: types/spotify-web-playback-sdk/index.d.ts [hunk 2]
```diff
@@ -64,23 +79,29 @@ declare namespace Spotify {
...
 
     interface PlaybackRestrictions {
-        disallow_pausing_reasons: string[];
-        disallow_peeking_next_reasons: string[];
-        disallow_peeking_prev_reasons: string[];
-        disallow_resuming_reasons: string[];
-        disallow_seeking_reasons: string[];
-        disallow_skipping_next_reasons: string[];
-        disallow_skipping_prev_reasons: string[];
+        disallow_pausing_reasons?: string[];
```
Comment (mraerino) [inline types/spotify-web-playback-sdk/index.d.ts:95]: could we use an enum here instead of a string? is the list of possible `reasons` known?

**Comment by mraerino** [comment]
the example data is great! just a few smaller questions

File: types/spotify-web-playback-sdk/index.d.ts (hunk 1 above)
Comment (aelsen) [reply types/spotify-web-playback-sdk/index.d.ts:127]: I agree, although unfortunately I can't find concrete documentation listing their values (same as the disallow_*_reasons), and the minified source for the playback SDK doesn't reveal anything either.

**Review by mraerino — APPROVED** [review]
i think this is fine to land, people can report issues if this ends up not being accurate

**Comment by aelsen** [comment]
Ready to merge

**Pull request merged**